sanic_ext/extensions/openapi/openapi.py

```diff
@@ -223,6 +233,18 @@ async def handler(request, *handler_args, **handler_kwargs):
 
 @overload
 def parameter(
+    *,
+    parameter: definitions.Parameter,
+    **kwargs,
+) -> Callable:
+    ...
+
+
+@overload
+def parameter(
+    name: None,
+    schema: None,
+    location: None,
     parameter: definitions.Parameter,
     **kwargs,
 ) -> Callable:
```

Comment:
Default args must follow non-default args, so I had to create a new overload in order for it to be backwards-compatible with both `@openapi.parameter(parameter=...)` and `@openapi.parameter(None, None, None, openapi.Parameter(...))` (no clue who in their sane mind would do this, but the option is there).